### ISSUE TITLE
command line utility should produce same errors for number fields as with HTML version [feature request]

### ISSUE THREAD

**Issue by tomasohara** [validator/validator#1858]
**Is your feature request related to a problem? Please describe.**

Yes, the command line check produces a misleading error report compared to HTML version.

1. The command line version indicates type=number not valid:

```
$ alias vnu='"${JAVA_HOME}/bin/java" -jar "./build/dist/vnu.jar"'

$ vnu --version
25.7.26

$ vnu bad-number-field.html
"file:/home/tomohara/programs/java/validator--nu-html/bad-number-field.html":9.7-9.85: error: Bad value “number” for attribute “type” on element “input”.
```

2. The HTML version correctly pinpoints the incorrect size and maxlength usage (via python ./checker.py run):

```
http://localhost:8888
...
1. Error: Attribute size not allowed on element [input](https://html.spec.whatwg.org/multipage/#the-input-element) at this point.

From line 9, column 7; to line 9, column 85
...
2. Error: Attribute maxlength is only allowed when the input type is email, password, search, tel, text, or url.

From line 9, column 7; to line 9, column 85
```

**Describe the solution you'd like**

The command line errors should match the HTML ones. As is, it appears that the command line is producing a false positive.

**Describe alternatives you've considered**

A bunch of web searching and checking source for tips.

**Additional context**

```
$ cat bad-number-field.html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta content="text/html; charset=UTF-8" http-equiv="content-type">
    <title>Bad Number Field</title>
  </head>
  <body>
    <label>#terms:
      <input id='num-terms-id' name='num-terms' maxlength='2' size='2' type='number'>
    </label>
  </body>
</html>


$ git log | head -5
commit 8cb7c369459b9a13260b08b80a70f3fd1066ca28
Author: sideshowbarker <mike@w3.org>
Date:   Sat Jul 26 14:11:46 2025 +0900

    Update tests
```

**Comment by sideshowbarker** [comment]
I can reproduce the inconsistent reporting behavior from the issue description, but at this point I have no idea what’s causing the inconsistency. Some further investigation is needed.

**Comment by sideshowbarker** [comment]
To clarify one thing: the _Bad value “number” for attribute “type” on element “input”_ error message that the CLI reports is essentially saying that the `type` attribute for an `input` element can’t have the value `number` when it has  a`size` attribute (or when it has a `maxlength` attribute — I don’t really know which case it’s internally checking first).

So that message isn’t incorrect — it’s just less helpful than the one the web-based service is emitting.

But the thing is, in the checker code, we don’t have an direct control over which of those different message cases ends up getting emitted. Instead we pass on the messages we get from the Jing backend that does the actual schema evaluation and validation.

So I don’t know what could be causing the Jing backend to be behaving differently in the CLI case from how it behaves in the web-based-service case. I can’t recall ever seeing this happen before.

**Comment by sideshowbarker** [comment]
Finally got this figured out and fixed. You can try with the latest jar.

If you ever happen to run into other cases, please raise new issues.

**Comment by tomasohara** [comment]
Works for me! Have a Happy New Year!
